src/test/java/org/folio/edge/sip2/repositories/DefaultResourceProviderTests.java

```diff
@@ -3,43 +3,60 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-import org.junit.jupiter.api.Test;
-
+import io.vertx.core.Vertx;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+@ExtendWith(VertxExtension.class)
 public class DefaultResourceProviderTests {
 
   @Test
-  public void canConstructDefaultConfigurationProvider(){
+  public void canConstructDefaultConfigurationProvider() {
     DefaultResourceProvider defaultConfigurationProvider = new DefaultResourceProvider();
     assertNotNull(defaultConfigurationProvider);
   }
 
   @Test
-  public void canRetrieveAcsConfiguration(){
+  public void canRetrieveAcsConfiguration(
+      Vertx vertx,
+      VertxTestContext testContext) {
     DefaultResourceProvider defaultConfigurationProvider = new DefaultResourceProvider();
-    JsonObject jsonConfig = defaultConfigurationProvider.retrieveResource(null);
-
-    assertNotNull(jsonConfig);
-
-    JsonObject acsConfig = jsonConfig.getJsonObject("acsConfiguration");
-
-    assertEquals("fs00000010test", acsConfig.getString("institutionId"));
-    assertEquals("1.23", acsConfig.getString("protocolVersion"));
+    defaultConfigurationProvider.retrieveResource(null).setHandler(
+        testContext.succeeding(jsonConfig -> testContext.verify(() -> {
```

Comment:
Do we need to use testContext.verify() here? Seems overkill...

Reply:
I think we do since we are dealing with futures here. Of course, in this case the futures should be complete so there is nothing asynchronous here. But I am not sure how the test would react if there was a failure in the `succeeding()` block with out the verify. I was following: [https://vertx.io/docs/vertx-junit5/java/#_use_any_assertion_library](https://vertx.io/docs/vertx-junit5/java/#_use_any_assertion_library). If you have a better way to handle `Future` based assertions, let me know.

Reply:
Let me checkout your branch to see if we can do something about it.

Reply:
Every test I see with a `succeeding()` uses `verify` when there are asserts in the block. I took out the verify and the test above still worked or failed based on the assert, but I'd be concerned to not follow this pattern because we know something about the logic behind `DefaultResourceProvider.retrieveResource()`.

While this test "works" without the verify, I tried in the similar test in the FOLIO provider. I added a 5 second sleep to the resource behind `retrieveResource` and the assertion fired, but the execution context was never notified, so the test failed due to timeout. With the `verify` in place the test fails correctly after 5 seconds. So, I think it is critical we follow this pattern even in simple cases like this where we use `Future` objects, but not asynchronously.

Does that make sense?

Reply:
OK, thanks for giving it more thoughts.